File: sortcolour.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        # TC O(n)
        # SC O(1)
        if nums is None or len(nums) == 0:
            return
        high = len(nums) -1
        low = 0
        mid = 0
        while mid <= high:
                
            if nums[mid] == 2:
                nums[mid],nums[high] = nums[high] ,nums[mid]
                high -= 1
            elif nums[mid] == 1:
                mid += 1
            else:
                nums[mid],nums[low] = nums[low] ,nums[mid]
                low +=1
                mid += 1
                

File: SortColour.java
// time complexity 0(n)
// space complexity 0(1)
class Solution {
    public void sortColors(int[] nums) {
        if(nums.length==0 || nums==null)
            return;
        int low=0;
        int high=nums.length-1;
        int mid =0;
        while(mid<=high)
        {
            if(nums[mid]==0)
            {
                swap(nums,mid,low);
                low++;
                mid++;
            }
            else if(nums[mid]==1)
            {
                mid++;
            }
            else
            {
                swap(nums,mid,high);
                high--;
            }
        }
    }
    private void swap(int[] nums, int i, int j)
    {
        int temp=nums[i];
        nums[i]=nums[j];
        nums[j]=temp;
    }
}